src/functions/env.ts

```diff
@@ -240,6 +252,62 @@ export function hasUserEnvs({
   return findEnvfiles(functionsSource, projectId, projectAlias, isEmulator).length > 0;
 }
 
+/**
+ * Write new environment variables into a dotenv file.
+ *
+ * Identifies one and only one dotenv file to touch using the same rules as loadUserEnvs().
+ * It is an error to provide a key-value pair which is already in the file.
+ * Not actually implemented yet.
+ */
+export function writeUserEnvs(toWrite: Record<string, string>, envOpts: UserEnvsOpts) {
+  const { functionsSource, projectId, projectAlias, isEmulator } = envOpts;
+  let envFiles = findEnvfiles(functionsSource, projectId, projectAlias, isEmulator);
+  if (envFiles.length === 0) {
+    createEnvFile(envOpts);
+    envFiles = findEnvfiles(functionsSource, projectId, projectAlias, isEmulator);
```

Comment:
nit* Maybe `createEnvFile` can return the name of the file it created. Then we can write:

```js
  envFiles = [createEnvFile(envOpts)]
```